p2p/net/swarm/swarm_transport.go

```diff
@@ -27,7 +27,10 @@ func (s *Swarm) TransportForDialing(a ma.Multiaddr) transport.Transport {
 		}
 		return nil
 	}
-	if isRelayAddr(a) {
+	if isProtocolAddr(a, ma.P_WEBRTC) {
+		return s.transports.m[ma.P_WEBRTC]
```

Comment:
The peer address is a circuitv2 address. So we need to check it before the circuitv2 check but the check is the same as the circuit v2 check. 

The address is of the form
`/ip4/1.2.3.4/udp/1/quic-v1/p2p/<relay-id>/p2p-circuit/webrtc`
So we need to ensure that we don't use the quic-v1 or the p2p-circuit transport for dialing.

Reply:
Yes but we don't want the circuit-v2 transport to handle dialing to the webrtc address. 
The previous code was 
if isRelayAddr => Use circuitv2 transport. 

But now we need
if isWebRTCAddr => Use webrtcprivate transport.
else if isRelayAddr => Use circuitv2 transport. 
This is because the webrtcprivate address has a p2p-circuit component but we want to use webrtcprivate transport. I could have written this as:
```
if isRelayAddr(addr):
	if isWebRTCAddr(addr):
		use webrtcprivate transport
	else
		use circuitv2 transport
```

The flow I've implemented is: 
peerAddr: `<relay-addr>/p2p-circuit/webrtc`

1. webrtcprivate.Transport.Dial is called with peerAddr
2. webrtcprivate.Transport.Dial adds the relay address inferred from the webrtcprivate address inferred from the peerstore. Then makes a host.Connect call to establish a relay connection with the peer. 
3. Once relay connection is established it opens a signaling stream and establishes a webrtc connection.

I've chosen this because it is similar to circuitv2 transport which first establishes a direct connection with the relay node and then a circuitv2 connection to the peer over the relayed connection. This also has the nice property that Transport.Dial can connect to the peer over the address no matter the state of the host when Transport.Dial is called. In this case, it means if no circuit-v2 connection exists, it'll make a circuit-v2 connection first.